pkg/registry/apis/provisioning/repository_controller.go

```diff
@@ -256,13 +276,13 @@ func (rc *RepositoryController) process(item *queueItem) error {
 			logger.InfoContext(ctx, "handle repository update")
 			status, err = repo.OnUpdate(ctx, logger)
 			if err != nil {
-				rc.logger.Error("OnUpdate", "error", err)
+				return fmt.Errorf("on update: %w", err)
```

Comment:
👍🏻  but the update functions should be careful to only error for unrecoverable issues

(eg, we should not error if we could not get the webhook status)

Reply:
why? it can return any error. It's up to us in queue to distinguish if it can be recovered or not based on known errors such us `ServiceIsUnavailable`.

Reply:
parked for: https://github.com/grafana/git-ui-sync-project/issues/49